pump/storage/storage.go

```diff
@@ -673,6 +674,7 @@ func (a *Append) doGCTS(ts int64) {
 		}
 	}()
 
+	l0TestLarge := 0
 	for {
 		nStr, err := a.metadata.GetProperty("leveldb.num-files-at-level0")
```

Comment:
Maybe just extract a function to check if further writes are going to cause compaction? Then we can simply mock the function in our test without introducing failpoint?

Reply:
I will use failpoint to inject error into `fsync`, here we intorduced failpoint is prepare for `fsync` test.